index.js

```diff
@@ -1,5 +1,111 @@
 const battleship = () => {
-  return 'The winner is...?'
+
+  //step 1
+
+  //objects of players(works)
+  let player1 = {
+    name: prompt("Player 1, enter your name").toUpperCase(),
+    shipCount: 4,
+    gameBoard:[
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ]]
+    }
+
+    let player2 = {
+    name: prompt("Player 2, enter your name").toUpperCase(),
+    shipCount: 4,
+    gameBoard:[
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ]]
+
+  }
+
+
+  //step 2(notice number of placed ships are not always consistent when ran. sometimes it shows 4 other times 3 maybe a two. not sure why. if you see what's wrong, please give me feedback. thanks!
+
+  //setup of ships
+
+    let ships;
+    let xPos;
+    let yPos;
+      for (i=0; i<player1.shipCount; i++){
```

Comment:
This loop needs to run not only only 4 times but as many times as is needed for each player to get 4 ships on their board (you might get a random x,y coordinate for a space that already has a ship, so the loop might need to run more than 4 times)

Reply:
@elizabethompson would the solution to your comment be to nest another for loop inside of this one? or is there a JavaScript method available that could help here? asking for a friend (me, lol)

Reply:
Yes so another student had the same issue and one student mentioned that maybe there needs to be a loop within that random loop to check if a ship is already there and if not put one and is so move on. I tried there suggestion but it still didnt; work. The logic you and the student metioned of why of not getting 4 ships all the time makes sense bc it may be going over the 1 if it's there and counting it as a new ship. im not sure how to fix. would i create a new loop or another loop within the same loop?
let ships;
  let xPos;
  let yPos;
    for (i=0; i<player1.shipCount; i++){
    xPos = Math.floor(Math.random() * 4);
    yPos = Math.floor(Math.random() * 4);
      while(player1.gameBoard[xPos][yPos]!==1){
      //if(parseInt(player1.gameBoard[xPos][yPos])!==1){
      xPos = Math.floor(Math.random() * 4);
      yPos = Math.floor(Math.random() * 4);
      
      player1.gameBoard[xPos][yPos] = 1;  
      ships++;
      }
    }
    //}

    for (i=0; i<player2.shipCount; i++){
    xPos = Math.floor(Math.random() * 4);
    yPos = Math.floor(Math.random() * 4);
      while(player2.gameBoard[xPos][yPos]!==1){
     // if(parseInt(player2.gameBoard[xPos][yPos])!==1){
      xPos = Math.floor(Math.random() * 4);
      yPos = Math.floor(Math.random() * 4);
        
      player2.gameBoard[xPos][yPos] = 1; 
      ships++;
      }
    }

Reply:
@Linda-Vu - @meljoi resolved this in her latest commit by changing the loop condition to run until the player has 4 ships instead of running 4 times only (it could take 4, 5, 10 iterations for the player to actually get 4 unique ship coordinates for their board).

Here's the code that she wrote:
```js
let ships=0;

while(ships < player1.shipCount) {
  ...
}
```

So to answer your question, the solution is not an additional loop, it's just using a different condition for the loop that's there already.